### PR TITLE
Add 7 new icons to weather card and forecast conditions

### PR DESCRIPTION
**Pull request by outon**
I have noticed that there are new icons in MDI relating to weather conditions.

I propose the inclusion of these new icons.

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @outon, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant-polymer#3726).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/cards/ha-weather-card.js [hunk 1]
```diff
@@ -284,12 +284,19 @@ class HaWeatherCard extends LocalizeMixin(EventsMixin(PolymerElement)) {
       exceptional: "hass:alert-circle-outline",
       fog: "hass:weather-fog",
       hail: "hass:weather-hail",
+      'hazy': "hass:weather-hazy",
```
Comment (iantrich) [inline src/cards/ha-weather-card.js]: Quotes not needed on hazy

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts [hunk 2]
```diff
@@ -49,12 +49,19 @@ const weatherIcons = {
   exceptional: "hass:alert-circle-outline",
   fog: "hass:weather-fog",
   hail: "hass:weather-hail",
+  'hazy': "hass:weather-hazy",
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-weather-forecast-card.ts]: Quotes not needed on hazy

**Review by iantrich — CHANGES_REQUESTED** [review]
Should be good after comments addressed

**Comment by Mariusthvdb** [comment]
could it be possible to also add the icon mdi:weather-night for condition clear-night?
see https://github.com/home-assistant/home-assistant/pull/26452#event-2613722551 where we added this for the dark sky sensor.

Not sure if this condition is returned in the weather forecast card conditions, but we do need a way to distinguish clear-night from a bright shining sun on the weather forecast cards.

**Comment by outon** [comment]
> great to see these added, thanks for that!
> could it be possible to also add the icon mdi:weather-night for condition clear-night?
> see [home-assistant/home-assistant#26452 (comment)](https://github.com/home-assistant/home-assistant/pull/26452#event-2613722551) where we added this for the dark sky sensor.

I see that your  [request](https://github.com/home-assistant/home-assistant/pull/26452#event-2613722551) seems to be already merged, probably on near future version will apear.

**Comment by bramkragten** [comment]
I think we can include these new icons, but I think more needs to be updated:

- The dev docs for supported states: https://developers.home-assistant.io/docs/en/entity_weather.html#recommended-values-for-state-and-condition
- Translations: https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/translations/en.json#L281

And we should have at least one weather component that uses these new states/icons.

**Comment by outon** [comment]
> * The dev docs for supported states: https://developers.home-assistant.io/docs/en/entity_weather.html#recommended-values-for-state-and-condition

I have created a pull request on that repository: https://github.com/home-assistant/developers.home-assistant/pull/315

> * Translations: https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/translations/en.json#L281

I have included all new weather conditions to translation file

> And we should have at least one weather component that uses these new states/icons.

I am creating a new AEMET custom component which will use them. I do not know about other official weather components. But I guess that once they have those new weather condition they will start using it.

**Review by balloob — CHANGES_REQUESTED** [review]
New weather conditions first need to be proposed and approved in the architecture repo: https://github.com/home-assistant/home-assistant/pull/

**Comment by MatthewFlamm** [comment]
I think the explosion of subtle variational themes should not be included.  It is too much of a burden to maintain when and if these icon names change or are removed. I would like to see some icons for different conditions like hurricane however.


https://github.com/home-assistant/architecture/issues/263

I started a similar request above recently to use for the recently merged nws integration. It didn't get much traction and `exceptional` wasn't supported at that time.
 Here are the list of unmapped conditions for nws integration that are currently in `exceptional`, as inspiration for meeting @balloob suggestion to have one integration support the icon, I.e. these could be remapped to a new icon. `partlycloudy-night` could also be supported fairly easily in nws and would be a good addition.

```
"Tornado",
"Hurricane conditions",
"Tropical storm conditions",
"Dust",
"Smoke",
"Haze",
"Hot",
"Cold"
```

**Comment by Mariusthvdb** [comment]
> New weather conditions first need to be proposed and approved in the architecture repo: https://github.com/home-assistant/home-assistant/pull/

would weather-night and weather-night-partly-cloudy be considered new conditions? Hope they are as old as night and day ;-) 
this shouldn't be possible/necessary at midnight:

Schermafbeelding 2019-09-20 om 23 43 59

this is more like it:

Schermafbeelding 2019-09-20 om 23 44 06

so please lets use:
Schermafbeelding 2019-09-20 om 23 48 33

**Comment by bramkragten** [comment]
Going to close this for now.

**Pull request closed without merging**